### DISCUSSION TITLE
Nextjs api routes giving 404 status on Vercel deployment

### DISCUSSION THREAD

**Discussion by Nickman87** [vercel/community#121] [Help] [answered]
Hi,

I have a pretty simple next.js application and I am using API calls with the built in API routes.

This works perfectly in development, but after deploying, all API routes return a 404 status, even though the call is correctly handled, the correct result is returned and the logs show the correct status being returned...
It seems like there is some gateway issue where the status is breaking?

example: https://plan2book-poc-firebase.vercel.app/api/getFood

If you call this in postman for example, you will get a 404, but it should be 200...

**Comment by mcsdevv** [comment]
Hello @Nickman87, that's definitely not expected.

Can you get in touch with [Vercel Support](https://vercel.com/support/request) with a full reproduction and the source code so that we can investigate?

**Comment by teug91** [comment, marked as answer]
I had this issue. Seemed liked Vercel got confused by there being two `api` directories. As soon as I renamed the `api` directory in root, the issue went away.

Before:
```
/api/
/pages/api/
```
After:
```
/lib/
/pages/api/
```

**Reply by Nickman87** [reply]
I also have a root api directory! Will test it out and get back to you.
Thanks for the tip!

**Reply by markozxuu** [reply]
@Nickman87  got positive results ?

**Reply by Nickman87** [reply]
@markozxuu Sorry, I completely lost track of this. I just verified and everything is working as expected. So Vercell has some issues it seems if your project also has an API folder :).

**Reply by keeganthomp** [reply]
Thank you! It would be great to mention this potential pitfall in the docs. Quite the time suck.

**Reply by sylvain-guehria** [reply]
Hollllllly thanks ! 2 days I am looking for the problem and it was a stupid folder name, thank you =D

**Reply by educobuci** [reply]
I had the same issue. Maybe they should include a warning message for this.

**Reply by equinusocio** [reply]
I still have the issue, but i have no folders called `api`...

**Reply by Pieparker** [reply]
Hey 2021ers, we appreciate you reaching out! Unfortunately, this topic is rather old and has been marked as Answered. If you’re exhibiting the same or a similar issue to the original poster, please feel free to create your own discussion and provide as much context as you can. This helps us to ensure that discussions are relevant, fresh and free of any possibly misleading context.

**Comment by loolooii** [comment]
I know this is from a long time ago, but I suddenly have this issue as well. It used to work correctly. `/api/{endpoint}` gives me `404` even though it exists.

**Reply by aledwassell** [reply]
Yeah I have this issue too, really odd.

**Reply by loolooii** [reply]
You need to remove ‘next export’ from your build command. Vercel uses your build command and ‘export’ is the default option. At least in my case it was.

**Reply by laniltee** [reply]
I don't have a `next export` command in the build. But I'm also seeing this error. It's giving default 404 error page of Next when an API route is called.

**Reply by laniltee** [reply]
The default demo `api/hello` route works perfectly. It only happens to newly added routes.

**Reply by andrejkaurin** [reply]
I am having the same issue

**Comment by allentofight** [comment]
+1, I have the same error

**Reply by amyegan** [reply]
Hey @allentofight. This topic is a few years old and has been marked as "Answered". Please create your own new discussion and provide as much context as you can. That will help us keep discussions organized and easy to follow 🙏
